Content.Server/GameObjects/Components/Items/Storage/StorageCounterComponent.cs

```diff
@@ -0,0 +1,82 @@
+﻿#nullable enable
+
+using System.Collections.Generic;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Tag;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Items.Storage
+{
+    /// <summary>
+    /// Storage that spawns and counts a single item.
+    /// Usually used for things like matchboxes, cigarette packs,
+    /// cigar cases etc.
+    /// </summary>
+    /// <code>
+    ///  - type: StorageCounter
+    ///    amount: 6 # Note: this field can be omitted
+    ///    countTag: Cigarette # Note: field doesn't point to entity Id, but its tag
+    /// </code>
+    [RegisterComponent]
+    public class StorageCounterComponent : Component
+    {
+        private string? _countTag;
+        private int? _maxAmount;
+
+        /// <summary>
+        /// Single item storage component usually have an attached StackedVisualizer.
+        /// </summary>
+        [ComponentDependency] private readonly AppearanceComponent? _appearanceComponent = default;
+
+        public override string Name => "StorageCounter";
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _countTag, "countTag", null);
```

Comment:
Also here slap a log warning if _countTag is null given the component is useless without it.